src/struct_mv/_hypre_struct_mv.h

```diff
@@ -2032,6 +2032,72 @@ hypre_BoxArrayBox(hypre_StructVectorDataSpace(vector), b)
 (hypre_StructVectorBoxData(vector, b) + \
  hypre_BoxIndexRank(hypre_StructVectorBox(vector, b), index))
 
+#endif
+/******************************************************************************
+ * Copyright 1998-2019 Lawrence Livermore National Security, LLC and other
+ * HYPRE Project Developers. See the top-level COPYRIGHT file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ ******************************************************************************/
+
+/******************************************************************************
+ *
+ * Header info for the struct matrix-matrix multiplication structures
+ *
+ *****************************************************************************/
+
+#ifndef hypre_STRUCT_MATMULT_HEADER
+#define hypre_STRUCT_MATMULT_HEADER
+
+#ifdef MAXTERMS
+#undef MAXTERMS
+#endif
+#define MAXTERMS 3
+
+/*--------------------------------------------------------------------------
+ * StructMMHelper data structure
+ *--------------------------------------------------------------------------*/
+
+/* product term used to compute the variable stencil entries in M */
+typedef struct hypre_StructMMhelper_struct
+{
+   hypre_StTerm    terms[MAXTERMS]; /* stencil info for each term */
+   HYPRE_Int       mentry;          /* stencil entry for M */
+   HYPRE_Complex   cprod;           /* product of the constant terms */
+   HYPRE_Int       types[MAXTERMS]; /* types of computations to do for each term */
+   HYPRE_Complex  *tptrs[MAXTERMS]; /* pointers to matrix data for each term */
+   HYPRE_Complex  *mptr;            /* pointer to matrix data for M */
+} hypre_StructMMHelper;
+
+/*--------------------------------------------------------------------------
+ * StructMMData data structure
+ *--------------------------------------------------------------------------*/
+
+typedef struct hypre_StructMMData_struct
```

Comment:
Nice job getting rid of the `void *` stuff.  I would call this `hypre_StructMatrixMatmultData_struct`.  It's long, but that's more a result of the term `StructMatrix` being long.  One thing we were doing in Struct before was to abbreviate to just `Struct` for `StructMatvec`, `StructMatmult`, etc.  Do something similar for the helper struct (replace `Data` with `Helper`).